cpp/log/tree_signer-inl.h

```diff
@@ -237,14 +237,12 @@ typename TreeSigner<Logged>::UpdateResult TreeSigner<Logged>::UpdateTree() {
   uint64_t min_timestamp = LastUpdateTime() + 1;
 
   // Sequence any new sequenced entries from our local DB.
+  auto it(db_->ScanEntries(cert_tree_->LeafCount()));
   for (int64_t i(cert_tree_->LeafCount());; ++i) {
     Logged logged;
-    typename Database<Logged>::LookupResult result(
-        db_->LookupByIndex(i, &logged));
-    if (result == Database<Logged>::NOT_FOUND) {
+    if (!it->GetNextEntry(&logged)) {
```

Comment:
Yeah; currently it'll `CHECK` if it hits a gap, but that's not entirely ideal given the potentially sparse nature of the data above the STH.